ddtrace/tracer.py

```diff
@@ -221,31 +252,32 @@ def configure(self, enabled=None, hostname=None, port=None, uds_path=None, https
             self.log.debug('Connecting to DogStatsd({})'.format(dogstatsd_url))
             self._dogstatsd_client = DogStatsd(**dogstatsd_kwargs)
 
-        if hostname is not None or port is not None or uds_path is not None or https is not None or \
-                filters is not None or priority_sampling is not None:
-            # Preserve hostname and port when overriding filters or priority sampling
-            # This is clumsy and a good reason to get rid of this configure() API
-            if hasattr(self, 'writer') and hasattr(self.writer, 'api'):
-                default_hostname = self.writer.api.hostname
-                default_port = self.writer.api.port
-                if https is None:
-                    https = self.writer.api.https
-            else:
-                default_hostname = self.DEFAULT_HOSTNAME
-                default_port = self.DEFAULT_PORT
-
-            self.writer = AgentWriter(
-                hostname or default_hostname,
-                port or default_port,
-                uds_path=uds_path,
-                https=https,
-                filters=filters,
-                priority_sampler=self.priority_sampler,
-                dogstatsd=self._dogstatsd_client,
-            )
+        # Copy existing value if needed
+        if hasattr(self, 'writer') and hasattr(self.writer, 'api'):
+            if hostname is None:
+                hostname = self.writer.api.hostname
 
-        # HACK: since we recreated our dogstatsd agent, replace the old write one
-        self.writer.dogstatsd = self._dogstatsd_client
+            if port is None:
+                port = self.writer.api.port
+
+            if https is None:
+                https = self.writer.api.https
+
+            if uds_path is None:
+                uds_path = self.writer.api.uds_path
+
+            if filters is None:
+                filters = self.filters
+
+        self.writer = AgentWriter(
```

Comment:
@jd What should we do with any existing writer thread when we are starting a new one in `configure()`?